src/emulator/functionsEmulator.ts

```diff
@@ -1082,11 +1093,15 @@ export class FunctionsEmulator implements EmulatorInstance {
     if (trigger) {
       const secrets: backend.SecretEnvVar[] = trigger.secretEnvironmentVariables || [];
       const accesses = secrets
-        .filter((s) => !secretEnvs[s.secret])
+        .filter((s) => !secretEnvs[s.key])
```

Comment:
Made these changes after discussing with Daniel. In CF3 world, secret and key are the same always, so the old code worked. However, based on what the fields of secretEnvVar represent, we  had these backwards before.